Backend/BlockChangesService.cs

```diff
@@ -57,10 +57,31 @@ public async Task<bool> CancelExchangeRequest(BlockChangeRequest request)
             return (a != null);
         }
 
-        private async Task AddAsync(BlockChangeRequest entityToAdd)
+        private async Task<BlockChangeRequest> AddAsync(BlockChangeRequest entityToAdd)
```

Comment:
The method should be renamed, it is not just add, may be: FindOrAdd...

Reply:
Method is renamed to FindOrAdd(...)